tensorflow-opgen/src/TensorFlow/OpGen.hs

```diff
@@ -237,40 +245,57 @@ quotedText = dquotes . strictText
 typeSig :: OpDef -> Doc
 typeSig d =
     foralls <+> constraints <+/>
-    signatureFold (mandatoryAttrInputs ++ tensorInputs ++ [outputs])
+    signatureFold (mandatoryAttrInputs ++ map snd tensorInputs ++ [outputs])
   where
-    foralls | Map.null typeMap = empty
+    foralls | null typeMap = empty
             | otherwise =
               "forall"
-              <+> sep (refTypes ++ map (strictText . fst) (Map.elems typeMap))
+              <+> sep (refVariableNames ++ typeMapTypeNames)
               <+> "."
-    constraints | Map.null typeMap = empty
+    typeMapTypeNames = map (strictText . fst) (Map.elems typeMap)
+    constraints | null typeMap = empty
                 | otherwise =
                   tuple (concatMap
                          (\(t, aDef) ->
                            "TensorType" <+> strictText t
                            : maybeToList (oneOfRestrictions aDef t))
                          (Map.elems typeMap)) <+> "=>"
     tensorInputs = zipWith tensorArg refTypes (d ^. inputArg)
-    refTypes = map (\x -> "v" <> int x) [1..length (d ^. inputArg)]
-    tensorArg refType arg = wrapArg refType arg <+>
-                            hang 0 ("-- ^" <+> argComment arg)
+    refTypes = ["v" <> int x | x <- [1..length (d ^. inputArg)]]
+    refVariableNames = catMaybes (map fst tensorInputs)
```

Comment:
Maybe move `refVariableNames` above `tensorInputs`? They were in a kind of reversed "dependency" order before and now it is a bit jumbled.